src/core/dev/net_device_table_mgr.cpp

```diff
@@ -420,31 +420,21 @@ void net_device_table_mgr::get_ip_list(local_ip_list_t &ip_list, sa_family_t fam
     m_lock.unlock();
 }
 
-int net_device_table_mgr::global_ring_poll_and_process_element(uint64_t *p_poll_sn_rx,
-                                                               uint64_t *p_poll_sn_tx,
-                                                               void *pv_fd_ready_array /*= NULL*/)
+bool net_device_table_mgr::global_ring_poll_and_process_element(uint64_t *p_poll_sn_rx,
+                                                                uint64_t *p_poll_sn_tx,
+                                                                void *pv_fd_ready_array /*= NULL*/)
 {
     ndtm_logfunc("");
-    int ret_total = 0;
+    bool all_drained = true;
 
     net_device_map_index_t::iterator net_dev_iter;
     for (net_dev_iter = m_net_device_map_index.begin();
          net_dev_iter != m_net_device_map_index.end(); net_dev_iter++) {
-        int ret = net_dev_iter->second->global_ring_poll_and_process_element(
+        all_drained &= net_dev_iter->second->global_ring_poll_and_process_element(
```

Comment:
Since all_drained = all_drained && poll_and_process() will not evaluate poll_and_process. And even for (poll_and_process() && all_drained) there is no clear guarantee to evaluate poll_and_process() with all modern optimizers. And since, the bit operation approach passes current known/supported compilers, i prefer to keep the bit op approach.